chasm/tree.go

```diff
@@ -214,36 +601,18 @@ func (n *Node) ApplySnapshot(
 	panic("not implemented")
 }
 
-func (n *Node) insert(
-	nodePath []string,
-	pNode *persistencespb.ChasmNode,
-	nodeBase *nodeBase,
-) {
-	if len(nodePath) == 0 {
-		n.persistence = pNode
-		return
-	}
-
-	childName := nodePath[0]
-	childNode, ok := n.children[childName]
-	if !ok {
-		childNode = newNode(nodeBase, n)
-		n.children[childName] = childNode
-	}
-	childNode.insert(nodePath[1:], pNode, nodeBase)
-}
-
 // IsDirty returns true if any node rooted at Node n has been modified.
 // The result will be reset to false after a call to CloseTransaction().
 func (n *Node) IsDirty() bool {
-	panic("not implemented")
+	return n.dirty
```

Comment:
this needs to recursively check all child nodes. The `dirty` flag is only for the current node, otherwise we can't compute mutations at the end of a transaction.